### PR TITLE
[SDAG][X86] Extend SplitVecOp_VSETCC for STRICT_FSETCC.

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

**Comment by FreddyLeaf** [comment]
This patch is to fix below backend crash:
https://gcc.godbolt.org/z/7cbTx7r1T

File: llvm/test/CodeGen/X86/vec-strict-cmp-512-skx.ll [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+; SKX-NEXT:    vpblendmd %ymm0, %ymm2, %ymm0 {%k2}
+; SKX-NEXT:    vpblendmd %ymm1, %ymm3, %ymm1 {%k1}
+; SKX-NEXT:    retq
+  %cond = call <16 x i1> @llvm.experimental.constrained.fcmp.v16f32(
+                                               <16 x float> %f1, <16 x float> %f2, metadata !"oeq",
+                                               metadata !"fpexcept.strict") #0
+  %res = select <16 x i1> %cond, <16 x i32> %a, <16 x i32> %b
+  ret <16 x i32> %res
+}
+
```
[the remark this answers is not in the document]
Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/vec-strict-cmp-512-skx.ll:20]: [7ff5d8d](https://github.com/llvm/llvm-project/pull/92509/commits/7ff5d8dbb0e9a6d4e2f64542fbfd3e352a7bbf6e)

File: llvm/test/CodeGen/X86/vec-strict-cmp-512-skx.ll [hunk 2]
```diff
@@ -0,0 +1,22 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -mtriple=x86_64 -mcpu=skx | FileCheck %s --check-prefixes=SKX
+
+; Test not crash for 512 targets but not prefer 512 vector width
```
[the remark this answers is not in the document]
Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/vec-strict-cmp-512-skx.ll]: [7ff5d8d](https://github.com/llvm/llvm-project/pull/92509/commits/7ff5d8dbb0e9a6d4e2f64542fbfd3e352a7bbf6e)

**Review by arsenm — APPROVED** [review]

**Review by RKSimon — APPROVED** [review]
LGTM

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**